pkg/querier/worker/util.go

```diff
@@ -146,8 +146,11 @@ func handleQueryRequest(ctx context.Context, request *queryrange.QueryRequest, h
 			}
 		}
 
+		// This block covers any errors that are not gRPC errors and will include all query errors.
```

Comment:
Is there a test covering this?

Reply:
Nope. That's from the hot fix. I'll add one.

Reply:
I've added a test but found that the code mapping was done by the [server.WriteError](https://github.com/grafana/loki/blob/main/pkg/util/server/error.go#L28) method.